panic: nil pointer dereference in CapacityReservationFromEC2 — lo.Ternary eagerly evaluates *cr.Interruptible

### Description

**Observed Behavior:**

Karpenter panics with `runtime error: invalid memory address or nil pointer dereference` in `CapacityReservationFromEC2` when reconciling an EC2NodeClass that has `capacityReservationSelectorTerms` configured. The panic occurs every reconciliation cycle, making capacity reservation discovery non-functional.

```
panic: runtime error: invalid memory address or nil pointer dereference
github.com/aws/karpenter-provider-aws/pkg/apis/v1.CapacityReservationFromEC2({...}, 0x36ce83ffb348)
    github.com/aws/karpenter-provider-aws/pkg/apis/v1/ec2nodeclass_status.go:264 +0x2ea
```

**Expected Behavior:**

Karpenter should safely handle capacity reservations where the `Interruptible` field is nil (which is the default for standard ODCRs and capacity blocks).

**Root Cause:**

At [`ec2nodeclass_status.go:264`](https://github.com/aws/karpenter-provider-aws/blob/v1.10.0/pkg/apis/v1/ec2nodeclass_status.go#L264):

```go
Interruptible: lo.Ternary(cr.Interruptible == nil, false, *cr.Interruptible),
```

`lo.Ternary` is **not lazy** — Go evaluates all function arguments before the call. `*cr.Interruptible` is dereferenced unconditionally, panicking when the pointer is nil.

**Fix:**

Replace with `lo.FromPtrOr`:

```go
Interruptible: lo.FromPtrOr(cr.Interruptible, false),
```

We have validated this fix on a patched v1.10.0 build — capacity reservations (both capacity blocks and open ODCRs) are discovered successfully after the change.

**Reproduction Steps:**

1. Have any capacity reservation in the account where `Interruptible` is nil (this is the default for standard ODCRs and capacity blocks)
2. Configure an EC2NodeClass with `capacityReservationSelectorTerms` that matches the reservation
3. Observe continuous panic in the nodeclass controller

**Versions:**
- Karpenter: v1.10.0 (also present on `main`, v1.11.0, v1.11.1 — same code)
- Kubernetes: 1.31
- EKS

--- a/pkg/apis/v1/ec2nodeclass_status.go
+++ b/pkg/apis/v1/ec2nodeclass_status.go
@@ -263,7 +263,7 @@ func CapacityReservationFromEC2(clk clock.Clock, cr *ec2types.CapacityReservatio
 		InstanceType:          *cr.InstanceType,
 		OwnerID:               *cr.OwnerId,
 		ReservationType:       reservationType,
-		Interruptible:         lo.Ternary(cr.Interruptible == nil, false, *cr.Interruptible),
+		Interruptible:         lo.FromPtrOr(cr.Interruptible, false),
 		State:                 state,
 	}, nil
 }

--- a/pkg/controllers/nodeclass/capacityreservation_test.go
+++ b/pkg/controllers/nodeclass/capacityreservation_test.go
@@ -131,8 +131,32 @@ var _ = Describe("NodeClass Capacity Reservation Reconciler", func() {
 					ReservationType:        ec2types.CapacityReservationTypeDefault,
 					Interruptible:          lo.ToPtr(true),
 				},
+				{
+					AvailabilityZone:       lo.ToPtr("test-zone-1a"),
+					InstanceType:           lo.ToPtr("m5.large"),
+					OwnerId:                lo.ToPtr(selfOwnerID),
+					InstanceMatchCriteria:  ec2types.InstanceMatchCriteriaTargeted,
+					CapacityReservationId:  lo.ToPtr("cr-nil-interruptible"),
+					AvailableInstanceCount: lo.ToPtr[int32](5),
+					State:                  ec2types.CapacityReservationStateActive,
+					ReservationType:        ec2types.CapacityReservationTypeDefault,
+					// Interruptible intentionally omitted (nil) — this is the default
+					// for standard ODCRs and capacity blocks returned by the EC2 API
+				},
 			},
 		})
+	})
+	It("should handle capacity reservations with nil Interruptible field", func() {
+		nodeClass.Spec.CapacityReservationSelectorTerms = append(nodeClass.Spec.CapacityReservationSelectorTerms, v1.CapacityReservationSelectorTerm{
+			ID: "cr-nil-interruptible",
+		})
+		ExpectApplied(ctx, env.Client, nodeClass)
+		ExpectObjectReconciled(ctx, env.Client, controller, nodeClass)
+		nodeClass = ExpectExists(ctx, env.Client, nodeClass)
+		Expect(nodeClass.StatusConditions().Get(v1.ConditionTypeCapacityReservationsReady).IsTrue()).To(BeTrue())
+		Expect(nodeClass.Status.CapacityReservations).To(HaveLen(1))
+		Expect(nodeClass.Status.CapacityReservations[0].ID).To(Equal("cr-nil-interruptible"))
+		Expect(nodeClass.Status.CapacityReservations[0].Interruptible).To(BeFalse())
 	})
 	It("should resolve capacity reservations by ID", func() {
 		const targetID = "cr-m5.large-1a-1"
@@ -206,10 +230,10 @@ var _ = Describe("NodeClass Capacity Reservation Reconciler", func() {
 		ExpectObjectReconciled(ctx, env.Client, controller, nodeClass)
 		nodeClass = ExpectExists(ctx, env.Client, nodeClass)
 		Expect(nodeClass.StatusConditions().Get(v1.ConditionTypeCapacityReservationsReady).IsTrue()).To(BeTrue())
-		Expect(nodeClass.Status.CapacityReservations).To(HaveLen(5))
+		Expect(nodeClass.Status.CapacityReservations).To(HaveLen(6))
 		Expect(lo.Map(nodeClass.Status.CapacityReservations, func(cr v1.CapacityReservation, _ int) string {
 			return cr.ID
-		})).To(ContainElements("cr-m5.large-1a-1", "cr-m5.large-1a-2", "cr-p5.48xlarge-1a", "cr-m5.large-1b-1", "cr-m5.large-1b-2"))
+		})).To(ContainElements("cr-m5.large-1a-1", "cr-m5.large-1a-2", "cr-p5.48xlarge-1a", "cr-m5.large-1b-1", "cr-m5.large-1b-2", "cr-nil-interruptible"))
 		for _, cr := range nodeClass.Status.CapacityReservations {
 			Expect(cr.InstanceMatchCriteria).To(Equal("targeted"))
 		}

